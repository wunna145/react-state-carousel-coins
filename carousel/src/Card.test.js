import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Card from './Card';

test('renders Card component without crashing', () => {
  const { getByText } = render(
    <Card caption="Test Caption" src="test.jpg" currNum={1} totalNum={3} />
  );

  const titleElement = getByText(/Test Caption/i);
  expect(titleElement).toBeInTheDocument();
});

test('renders Card component correctly', () => {
    const tree = renderer
      .create(<Card caption="Test Caption" src="test.jpg" currNum={1} totalNum={3} />)
      .toJSON();
  
    expect(tree).toMatchSnapshot();
});