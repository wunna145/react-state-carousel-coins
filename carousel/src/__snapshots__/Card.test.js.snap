// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card component correctly 1`] = `
<div
  className="Card"
>
  <h4
    className="Card-title"
  >
    Test Caption
  </h4>
  <img
    alt="Test Caption"
    className="Card-image"
    src="test.jpg"
  />
  <small
    className="Card-small"
  >
    Image 
    1
     of 
    3
    .
  </small>
</div>
`;
